library/src/main/java/com/alamkanak/weekview/WeekView.java

```diff
@@ -218,6 +224,8 @@ public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float ve
             if (mIsZooming)
                 return true;
 
+            mCurrentFlingDirection = mCurrentScrollDirection;
+
             if ((mCurrentFlingDirection == Direction.LEFT && !mHorizontalFlingEnabled) ||
```

Comment:
I thinks is necessary add a new validation here:

`if ((mCurrentFlingDirection == Direction.LEFT && !mHorizontalFlingEnabled || !mHorizontalScrollEnabled) ||
                    (mCurrentFlingDirection == Direction.RIGHT && !mHorizontalFlingEnabled || !mHorizontalScrollEnabled) ||
                    (mCurrentFlingDirection == Direction.VERTICAL && !mVerticalFlingEnabled || !mVerticalScrollEnabled)) {
                return true;
            }`

Reply:
Where exactly ?